src/components/modal.js

```diff
@@ -0,0 +1,24 @@
+import A11yDialog from "a11y-dialog";
+
+/**
+ *@param {string } modalClass
+ * @param {string} contentId
+ * @param {function} onShow
+ */
+const init = (modalClass, contentId, onShow = () => {}) => {
+
+  const modalClasses = document.querySelectorAll(`[class*=${modalClass}]`);
+
+  modalClasses.forEach(e => {
```

Comment:
can we change the `e` to something like `modalElm`?

Reply:
this will make it more readable.

Reply:
another nice to have, but not required is to move the processing of each modal to a function.

what that would look like is.

```js
// This is somehwere above your init method
const initalizeModal = (modalElm) => {
  const dialog = document.getElementById(modalElm.id);
  const content = document.getElementById(contentId);
  const modal = new A11yDialog(dialog, content);
  modal.on("show", onShow);

  return modal;
}
```

Then your current code would change to this

```js
modalElms.forEach(initalizeModal);
```

This makes the code a bit more reusable.

Reply:
done